### DISCUSSION TITLE
Change auth docs

### DISCUSSION THREAD

**Discussion by olegKusov** [vercel/next.js#23908] [Ideas]
### Describe the feature you'd like to request

Add examples in documentation with custom backend servers. For example, I created WithAuth function. And spent 2 weeks for research. It would be faster if the documentation was better.

### Describe the solution you'd like

I wanna get normal documentation for auth.

### Describe alternatives you've considered

![image](https://user-images.githubusercontent.com/28058268/114278484-bde9e300-9a38-11eb-8f74-df29b309cffb.png)

First, I thought that I can once get my access and refresh tokens, pass them to react Context in getServerSideProps and then use this useContext data everywhere. BUT... NextJs is not about it. I can't pass react contenxt inside getServerSideProps. Sooo, then I started my research. And after almost 2 weeks I found, that I need to send on every page request to refresh-token endpoint, because with out it I can't create normal and secure authentication.

So know I want from you little thing. Please, add example to documentation with principles of data flow in NextJS. Because maybe for you it will be clear, but for new people it's not clear. Give us more examples, tell us about data flow and about usage of useContext inside getServerSideProps. For noobs it's not clear, really. Give us authentication examples with custom backends without auth.js library or something else - just nextjs.

I really hate nextjs docs because I spent a lot of time for research. Now for me of course it's clear how it works , but at first It was not clear. I used stackoverflow for this result. But why I need to spend my time for research and surfing questions on stackoverflow, if I can just simply read normal documentation, with normal examples, with some animations of data flow or something else. Thank you.

**Comment by daniellin215** [comment]
Hey @olegKusov, I agree with you. I spent a lot of time researching but ultimately went down the path of nextauth.js and going serverless... I feel that my backend is boxed in at this point, because I cannot make/customize my own onboarding flow for my app because of the limitations of the backend.

Did you also skip nextauth because you have your own backend server? And was this `withAuth` needed to protect pages that require the user to be logged in?

**Comment by GabenGar** [comment]
There is no such thing as "normal auth documentation" because authentication is intertwined with cryptography and therefore full of mystic knowledge (jwt being one of them).
